lib/pure/nativesockets.nim

```diff
@@ -125,18 +127,26 @@ proc toInt*(p: Protocol): cint
 when not useWinVersion:
   proc toInt(domain: Domain): cshort =
     case domain
+    of AF_UNSPEC:      result = posix.AF_UNSPEC.cshort
     of AF_UNIX:        result = posix.AF_UNIX.cshort
     of AF_INET:        result = posix.AF_INET.cshort
     of AF_INET6:       result = posix.AF_INET6.cshort
-    else: discard
+
+  proc toKnownDomain*(family: cint): Option[Domain] =
```

Comment:
I'm going to link to my comment here again because GitHub decided to hide it:

https://github.com/nim-lang/Nim/pull/5763#discussion_r113963464